input/postgres/relations.go

```diff
@@ -162,7 +162,7 @@ SELECT relid,
 `
 
 const viewDefinitionSQL string = `
-	WITH locked_relids AS (SELECT DISTINCT relation relid FROM pg_catalog.pg_locks WHERE mode = 'AccessExclusiveLock' AND relation IS NOT NULL)
+	WITH locked_relids AS (SELECT DISTINCT relation relid FROM pg_catalog.pg_locks WHERE mode = 'AccessExclusiveLock' AND relation IS NOT NULL AND locktype <> 'tuple')
```

Comment:
I was not actually able to generate a lock like this on the view itself. When I tried conflicting `SELECT FOR UPDATE`s on a view, the waiting query got a locktype = 'tuple' AccessExclusiveLock on the underlying table instead. We could drop this, although based on the other behavior, I think this is safe.